src/tests/Logary.Facade.Tests/Facade.fs

```diff
@@ -62,33 +79,125 @@ let tests =
       let logger = Targets.create level
       Message.event Verbose "Hi {name}" |> logger.logSimple
 
-    testCase "extract matches" <| fun _ ->
-      let str = "Hi {ho}, we're { something going on  } special"
+    testCase "extract invalid property tokens as text" <| fun _ ->
+      let template = "Hi {ho:##.###}, we're { something going on } special"
       let fields = Map [ "ho", box "hola"; "something going on", box "very" ]
-      let subject = Formatting.extractMatches fields str
-      Expect.equal subject.Length 2 "Should match two things"
-      Expect.equal (subject |> List.map (fun (name, _, _) -> name))
-                   ["ho"; "something going on"]
-                   "Should extract correct names"
-      Expect.equal (subject |> List.map (fun (_, _, m) -> m.Value))
-                   ["{ho}"; "{ something going on  }"]
-                   "Should extract correct match values"
-
-    testCase "extract matches evil" <| fun _ ->
-      let strEvil = "{a}{b}{  c}{d  } {e} fghij {k}} {l} m {{}}"
-      let aThroughL =['a'..'l'] |> List.map string 
-      let fields = aThroughL |> List.map (fun s -> s, s) |> Map.ofList
+      let tokens = parseTemplateTokens template
+      Expect.equal tokens.Length 5 "Extracts 4 texts parts and 1 property part"
+      Expect.equal tokens
+                   [
+                    TextToken ("Hi ")
+                    PropToken ("ho", "##.###")
+                    TextToken (", we're ") // <- a quirk of the parser
+                    TextToken ("{ something going on }")
+                    TextToken (" special")
+                   ]
+                   "Should extract correct name and format parts"
+
+    testCase "handles evil property strings correctly" <| fun _ ->
+      let template = "{a}{b}{  c}{d  } {e} fghij {k}} {l} m {{}}"
+      // validity:     +  +  --    --   +         +    +     --
+      let expectedPropertyNames = ["a"; "b"; "e"; "k"; "l"]
       let strEvilRes = "abcd e fghij k} l m {{}}"
-      let subject = Formatting.extractMatches fields strEvil
-      Expect.equal subject.Length 7 "Should match 7 things, a-l, not m and not {}"
-      Expect.equal (subject |> List.map (fun (name, value, _) -> name))
-                   ["a"; "b"; "c"; "d"; "e"; "k"; "l"] 
+      let subject = parseTemplateTokens template
+      Expect.equal (subject |> List.choose (function PropToken (n,f) -> Some n | _ -> None))
+                   expectedPropertyNames
                    "Should extract relevant names"
-      Expect.equal (subject |> List.map (fun (_, _, m) -> m.Index))
-                   [0; 3; 6; 11; 17; 27; 32]
-                   "Should extract correct indicies"
 
-    testCase "format template properly" <| fun _ ->
+    testCase "literate tokenizes with field names correctly" <| fun _ ->
+      let str = "Added {item} to cart {cartId} for {loginUserId} who now has total ${cartTotal}"
```

Comment:
No, in this case, $ was part of text, not part of the property. One could make a fancy .net format string for currency though, maybe I should add a test for that.